src/mito2/src/worker/handle_alter.rs

```diff
@@ -47,6 +47,15 @@ impl<S> RegionWorkerLoop<S> {
 
         // Get the version before alter.
         let version = region.version();
+        if version.metadata.schema_version > request.schema_version {
```

Comment:
We check the schema version here
https://github.com/GreptimeTeam/greptimedb/blob/14e6998d4160b9f6d1fe4fd0a9cb27bb13a92c9f/src/store-api/src/region_request.rs#L191-L200